en/flight_stack/controller_diagrams.md

```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
 * Depending on the mode, the outer (position) loop is bypassed (shown as a multiplexer after the outer loop). The position loop is only used when holding position or when the requested velocity in an axis is null.
 * The integrator in the inner loop (velocity) controller includes an anti-reset windup (ARW) using a clamping method.
 
+## Fixed-Wing Position Controller
+
+### Total Energy Control System (TECS)
+The PX4 implementation of the Total Energy Control System (TECS) enables simultaneous control of true airspeed and altitude of a fixed wing aircraft.
+The code is implemented as a library which is used in the fixed wing position control module (make a reference here). As seen in the diagram below, TECS receives as inputs airspeed and altitude setpoints and outputs a throttle and pitch angle setpoint. These two outputs are sent to the fixed wing attitude controller which implements the attitude control solution. It's therfore important to understand that the performance of TECS is directly affected by the performance of the pitch control loop. A poor tracking of airspeed and altitude is often caused by a poor tracking of the aircraft pitch angle.
```

Comment:
Missing reference "(make a reference here)" - where should this go?

Reply:
I'd be tempted to move the diagram to just before your text "As seen in the diagram below," (and of course changing that from "below" to "above". This effectively shows your point, which is that TECS is part of the Fixed Wing Position control module.

Reply:
Typo "tranforming "